src/views/Sale/components/PurchaseTokensForm/index.tsx

```diff
@@ -166,50 +172,91 @@ export const PurchaseTokensForm = ({ saleId }: PurchaseTokensFormComponentProps)
+
+    if (!tokenQuantity) {
+      return console.error('token amount == null')
+    }
+
+    const fixedPriceSaleContract = FixedPriceSale__factory.connect(sale.id, getProviderOrSigner(library, account))
+
+    // Update state
+    setTxPending(true)
+    // Convert tokenQuantity (number) to 18-decimal BigNumber
+    // Sign and send transaction
+    fixedPriceSaleContract
+      .buyTokens(utils.parseEther(tokenQuantity.toString()))
+      .then(tx => tx.wait(1)) // wait one network confirmation
+      .then(receipt => {
+        console.log(receipt)
+      })
+      .catch(error => {
+        console.error(error)
+        toast.error(t('errors.purchase'))
+      })
+      .then(() => {
+        toast.success(t('success.purchase'))
+        setTxPending(false)
+      })
+  }, [account, library, sale, tokenQuantity, t])
+
   /**
    * Handles the form submission
    * @param event
    */
-
   const onSubmit = useCallback(
     async (event: FormEvent<HTMLFormElement>) => {
       event.preventDefault()
-      console.log('Buying tokens')
 
-      if (!sale) {
-        return console.error('no sale')
+      if (!sale || !purchaseValue) return null
+
+      if (sale.minimumRaise > BigNumber.from(0)) {
+        const totalSupply = formatBigInt(sale.sellAmount, sale.tokenOut.decimals)
+        const threshold = (formatBigInt(sale.minimumRaise) * 100) / totalSupply
+        const totalAmountPurchased = totalPurchased(totalBids)[0].amount
+        const amountDisplayed = Number(
+          ethers.utils.formatUnits(totalAmountPurchased, sale.tokenOut.decimals).slice(0, 5)
+        )
+        const percentageSold = (amountDisplayed / totalSupply) * 100
+
+        if (percentageSold < threshold) {
+          toggleConfirmation()
+          return
+        }
       }
 
-      if (!account || !library) {
-        return console.error('no connected signer')
-      }
-
-      if (!tokenQuantity) {
-        return console.error('token amount == null')
-      }
-
-      const fixedPriceSaleContract = FixedPriceSale__factory.connect(sale.id, getProviderOrSigner(library, account))
-
-      // Update state
-      setTxPending(true)
-      // Convert tokenQuantity (number) to 18-decimal BigNumber
-      // Sign and send transaction
-      fixedPriceSaleContract
-        .buyTokens(utils.parseEther(tokenQuantity.toString()))
-        .then(tx => tx.wait(1)) // wait one network confirmation
-        .then(receipt => {
-          console.log(receipt)
-        })
-        .catch(error => {
-          console.error(error)
-          toast.error(t('errors.purchase'))
-        })
-        .then(() => {
-          toast.success(t('success.purchase'))
-          setTxPending(false)
-        })
+      purchaseTokens()
     },
-    [sale, library, account]
+    [toggleConfirmation, sale, totalBids, totalPurchased, purchaseTokens, purchaseValue]
+  )
+
+  const confirmationModalContent = (
+    <div>
+      <p>{t('texts.saleUnderSoftCap')}</p>
+
+      <p>
+        <Trans
```

Comment:
Is there a reason for using the Trans component over the useTranslation hook?
Interpolation should work with the hook:
https://www.i18next.com/translation-function/interpolation#basic

Reply:
Oh, I somehow completely forgot this exists 🤦‍♂️ Fixing now.

Reply:
Nice, everything else looks good to me